minifi-c2/minifi-c2-commons/src/main/java/org/apache/nifi/minifi/c2/model/AgentInfo.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.c2.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Collection;
+
+@ApiModel
+public class AgentInfo {
+
+    private String identifier;
+    // TODO, do we also need identity. e.g., cert DN
+    private String agentClass;
+    private Collection<String> tags;
+    private AgentManifest agentManifest;
+    private AgentStatus status;
+    // private Map<String, String> configProperties;  // TODO we should add this information eventually, but we need to handle the best way to handle sharing sensitive properties.
+
+    @ApiModelProperty(
+            value = "A unique identifier for the Agent",
+            notes = "Usually set when the agent is provisioned and deployed")
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+
+    @ApiModelProperty(
+            value = "The class or category label of the agent, e.g., 'sensor-collector'",
+            notes = "Usually set when the agent is provisioned and deployed")
+    public String getAgentClass() {
+        return agentClass;
+    }
+
+    public void setAgentClass(String agentClass) {
+        this.agentClass = agentClass;
+    }
+
+    @ApiModelProperty("Arbitrary tags that describe the agent in addition to the class")
+    public Collection<String> getTags() {
+        return tags;
```

Comment:
can this be an immutable collection ? Preferably a set since I imagine similar tags have no real meaning?

Reply:
Good idea, will do. I'll review all getters to see which ones should return immutable collections.